src/main/java/teammates/ui/controller/StudentProfileEditSaveAction.java

```diff
@@ -65,7 +74,7 @@ private StudentProfileAttributes extractProfileData() {
     private void preprocessParameters(StudentProfileAttributes studentProfile) {
         studentProfile.shortName = StringHelper.trimIfNotNull(studentProfile.shortName);
         studentProfile.email = StringHelper.trimIfNotNull(studentProfile.email);
-        studentProfile.gender = StringHelper.trimIfNotNull(studentProfile.gender);
+        studentProfile.gender = Gender.valueOf(StringHelper.trimIfNotNull(studentProfile.gender.name()));
```

Comment:
This is redundant because you're transforming an enum to string and back to enum again.